### PR TITLE
publish the file name as SemVer

### PR DESCRIPTION
**Pull request by ZhengzheYang**
Update to publish the files with name specified [here](https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/blob/master/docs/versioning.md#tldr). Addressing issue #476

### CODE REVIEW

File: scripts/asm-installer/release_asm_installer [hunk 1]
```diff
@@ -133,18 +143,28 @@ publish_script() {
   local BRANCH_NAME; BRANCH_NAME="${1}"
   local VERSION; VERSION="${2}"
   local SCRIPT_NAME; SCRIPT_NAME="${3}"
-  local FILE_NAME; FILE_NAME="${SCRIPT_NAME}_${VERSION}"
-  local FILE_PATH; FILE_PATH="${BUCKET_PATH}/${FILE_NAME}"
-  local FILE_URI; FILE_URI="${BUCKET_URL}/${FILE_PATH}"
+  local TAG;
```
Comment (zerobfd) [inline scripts/asm-installer/release_asm_installer:146]: These happen to be git tags, but they're versions first. (i.e. we name the tag after the version, not the other way around)

Can we change TAG to STABLE_VERSION?

**Review by zerobfd — CHANGES_REQUESTED** [review]
The logic looks good. I have a couple of style nits.

File: scripts/asm-installer/release_asm_installer [hunk 2]
```diff
@@ -113,6 +112,17 @@ check_tags() {
...
 
+get_tags() {
+  local VER; VER="$(./"${SCRIPT_NAME}" --version || true)"
+  IFS="+" read -r VERSION CONFIG <<EOF
+${VER}
+EOF
+
+  cat << EOF
+${VERSION}-${CONFIG}
+EOF
```
[the remark this answers is not in the document]
Comment (ZhengzheYang) [reply scripts/asm-installer/release_asm_installer:123]: Done

File: scripts/asm-installer/release_asm_installer [hunk 3]
```diff
@@ -113,6 +112,17 @@ check_tags() {
   fi
 }
 
+get_tags() {
+  local VER; VER="$(./"${SCRIPT_NAME}" --version || true)"
+  IFS="+" read -r VERSION CONFIG <<EOF
+${VER}
+EOF
```
[the remark this answers is not in the document]
Comment (ZhengzheYang) [reply scripts/asm-installer/release_asm_installer:119]: Got it. Never seen this one before.

File: scripts/asm-installer/release_asm_installer (hunk 1 above)
Comment (ZhengzheYang) [reply scripts/asm-installer/release_asm_installer:146]: Done

File: scripts/asm-installer/release_asm_installer [hunk 4]
```diff
@@ -113,6 +112,12 @@ check_tags() {
   fi
 }
 
+get_stable_version() {
+  local VER; VER="$(./"${SCRIPT_NAME}" --version || true)"
+
+  echo ${VER//+/-}
```
Comment (zerobfd) [inline scripts/asm-installer/release_asm_installer:118]: Quote the ${}: https://github.com/koalaman/shellcheck/wiki/SC2046

File: scripts/asm-installer/release_asm_installer [hunk 5]
```diff
@@ -133,18 +138,28 @@ publish_script() {
...
 
   if [[ ! -f "${SCRIPT_NAME}" ]]; then echo "${SCRIPT_NAME} not found" >&2; return; fi
 
-  if ! changes_necessary "${FILE_URI}" "${FILE_NAME}"; then return; fi
-
   check_tags "${BRANCH_NAME}" "${SCRIPT_NAME}"
 
+  STABLE_VERSION="$(get_stable_version)"
+  write_and_upload ${SCRIPT_NAME} ${VERSION}
+  write_and_upload ${SCRIPT_NAME} ${STABLE_VERSION}
```
Comment (zerobfd) [inline scripts/asm-installer/release_asm_installer:151]: Same as above: https://github.com/koalaman/shellcheck/wiki/SC2046

**Review by zerobfd — APPROVED** [review]
Add quotes then lgtm. Thanks!

File: scripts/asm-installer/release_asm_installer (hunk 4 above)
Comment (ZhengzheYang) [reply scripts/asm-installer/release_asm_installer]: Done. Will review the shellcheck rules comprehensively.

File: scripts/asm-installer/release_asm_installer (hunk 5 above)
Comment (ZhengzheYang) [reply scripts/asm-installer/release_asm_installer]: Done

**Pull request merged**